features/events/syslog/src/test/java/org/opennms/netmgt/syslogd/ConvertToEventTest.java

```diff
@@ -431,4 +438,49 @@ public void testHostNameProvided() {
         assertNotNull(eventHostname);
         assertEquals(hostname, eventHostname);
     }
+
+    /**
+     * Make sure we use the received timestamp if there is no timestamp included in the syslog message.
+     */
+    @Test
+    public void testNoTimestampInMessage() {
+        RadixTreeParser oldRadixParser = RadixTreeSyslogParser.getRadixParser();
+
+        // Create a custom radix tree parser that can handle our message that contains no timestamp
+        RadixTreeParser radixParser = new RadixTreeParser();
+        radixParser.teach(GrokParserStageSequenceBuilder.parseGrok("%{STRING:message}").toArray(new ParserStage[0]));
+        RadixTreeSyslogParser.setRadixParser(radixParser);
+
+        String syslogMessage = "%CDP-4-NATIVE_VLAN_MISMATCH: Native VLAN mismatch discovered on GigabitEthernet5/9 " +
+                "(75), with switch.fqdn.com GigabitEthernet2/4/21 (2)";
+        Date now = new Date();
+        Event event = parseSyslog("testNoTimestampInMessage", radixConfig, syslogMessage, now);
+        RadixTreeSyslogParser.setRadixParser(oldRadixParser);
+
+        Date eventTime = event.getTime();
+        assertNotNull(eventTime);
+        assertEquals(eventTime, now);
```

Comment:
The arguments here should be flipped expected is `now`, actual is `eventTime`. The `assertNotNull` above is also redundant, if now.equals(eventTime) then eventTime cannot be null.